### PR TITLE
Hsd8 551 anchor and email links

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Change the Link widget in the WYSIWYG:
  * Allow anchors to be easily linked to
  * Allow emails to be easier to create
  * Integrate the new widget with linkit module to suggest page paths.

# Needed By (Date)
- End of the sprint 1/23

# Urgency
- High

# Steps to Test

1. checkout this branch
1. composer update --prefer-source
1. Sync a site `blt drupal:sync --site=[sitename]`
1. create any piece of content
1. in the WSYIWYG add some text, and click to create a link.
1. validate when typing in the url field, page suggestions populate
1. validate you can quickly and easily create a link to an anchor within the wysiwyg
1. validate you can quickly and easily create a mailto: link within the same link widget.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: config/default/filter.format.basic_html.yml [hunk 1]
```diff
@@ -43,7 +43,7 @@ filters:
...
+      allowed_html: '<em> <strong> <cite> <blockquote cite> <code> <ul type> <ol start type> <li> <dl> <dt> <dd> <h2 id class> <h3 id class> <h4 id class> <h5 id class> <h6 id class> <img src alt data-entity-type data-entity-uuid data-align data-caption class> <drupal-entity data-entity-type data-entity-uuid data-entity-embed-display data-entity-embed-display-settings data-align data-caption data-embed-button class> <table class> <caption class> <tbody class> <thead> <tfoot> <th scope class> <td class> <tr> <a id href name target hreflang data-entity-substitution data-entity-type data-entity-uuid title class> <p class> <div class> <br> <span class><i class>'
```
Comment (sherakama) [inline config/default/filter.format.basic_html.yml]: missing space between `<span>` and `<i>`

File: composer.json [hunk 2]
```diff
@@ -57,11 +57,43 @@
...
+        },
+        "require": {
+          "composer/installers": "~1.0"
+        }
+      }
+    },
+    {
+      "type": "package",
+      "package": {
+        "name": "ckeditor-plugin/fakeobjects",
```
Comment (sherakama) [inline composer.json:79]: What needs this library? What are you faking in the editor?

File: config/default/editor.editor.basic_html.yml [hunk 3]
```diff
@@ -28,8 +28,9 @@ settings:
         -
           name: Links
           items:
-            - DrupalLink
-            - DrupalUnlink
+            - Link
```
Comment (sherakama) [inline config/default/editor.editor.basic_html.yml:31]: Was this change caused by a module update?

File: docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.profile [hunk 4]
```diff
@@ -17,6 +17,15 @@ use Drupal\user\Entity\User;
 use Drupal\user\Entity\Role;
 use Drupal\user\RoleInterface;
 
+/**
+ * Implements hook_ckeditor_plugin_info_alter().
+ */
+function su_humsci_profile_ckeditor_plugin_info_alter(array &$definitions) {
+  if (isset($definitions['drupalfontawesome'])) {
+    $definitions['drupalfontawesome']['class'] = 'Drupal\su_humsci_profile\Plugin\CKEditorPlugin\HumsciFontAwesome';
```
Comment (sherakama) [inline docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.profile]: All this to alter a button? Damn. Be aware by doing this you (the project) are now responsible for keeping this class up to date.

File: composer.json (hunk 2 above)
Comment (pookmish) [reply composer.json:79]: its needed by the anchor link module https://www.drupal.org/project/anchor_link
not entirely sure what is does

File: config/default/editor.editor.basic_html.yml (hunk 3 above)
Comment (pookmish) [reply config/default/editor.editor.basic_html.yml:31]: Swapping the core DrupalLink for the link button provided in https://www.drupal.org/project/anchor_link

File: docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.profile (hunk 4 above)
Comment (pookmish) [reply docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.profile]: Yeah. its really kinda difficult and annoying. I hated having to do it. but without changing the icon, there are two flag icons. one for anchor link and another for font awesome. Maybe theres another way :man_shrugging:

Comment (sherakama) [reply docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.profile]: how about with some straight up CSS.

screen shot 2019-01-15 at 9 29 41 am

Comment (pookmish) [reply docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.profile]: well why didnt i see that before. i'll give it a shot

Comment (pookmish) [reply docroot/profiles/humsci/su_humsci_profile/su_humsci_profile.profile]: good idea. luckily i had recently worked on CKEditor api with the patch that is included in the PR

**Comment by pookmish** [comment]
Back up for review. I changed the method of replacing the icon. Instead of overridding the plugin, i created a new plugin that simply changes the icon after the ckeditor is ready.

**Review by sherakama — APPROVED** [review]
Ooo. I like this much better. Thank you for that refactor. This will help keep maintenance down.

**Pull request merged**